embark-anyone.lua

```diff
@@ -0,0 +1,108 @@
+local dialogs = require 'gui.dialogs'
+local choices = {}
+
+function addCivToEmbarkList(info)
+   local viewscreen = dfhack.gui.getDFViewscreen(true)
+
+   viewscreen.start_civ:insert ('#', info.civ)
+   viewscreen.start_civ_nem_num:insert ('#', info.nemeses)
+   viewscreen.start_civ_entpop_num:insert ('#', info.pops)
+   viewscreen.start_civ_site_num:insert ('#', info.sites)
+end
+
+function embarkAnyone()
+   local viewscreen = dfhack.gui.getDFViewscreen(true)
+
+
+   if viewscreen._type ~= df.viewscreen_choose_start_sitest then
+      qerror("This script can only be used on the embark screen!")
+   end
+
+   for i, civ in ipairs (df.global.world.entities.all) do
+      if civ.type == df.historical_entity_type.Civilization then
```

Comment:
instead of placing the entire body of the for loop in an if statement, consider inverting the logic and skipping the for body (that is, a "continue" statement in other languages) using a Lua label:
```lua
for i, civ in ipairs(df.global.world.entities.all) do
    if civ.type ~= df.historical_entity_type.Civilization then goto continue end
    ...
    ::continue::
end
```
This reduces the overall indentation and can make the code easier to reason about and follow

Reply:
Ahhhh okay

Sorry, I'm still used to C, where using goto is _heavily_ frowned upon.

Reply:
totally understood. I had to get used to this Lua-ism as well